### DISCUSSION TITLE
A question on Conversation IDs and how to set them

### DISCUSSION THREAD

**Discussion by Jamesjjvr** [acheong08/ChatGPT#411] [Q&A]
Terribly sorry to bother you, but after looking around i cant seem to find an answer. An issue I face is that whenever something happens and I am forced to rerun the code, I face the problem of the code starting a new chat instead of using the previous one. I am aware of the existence of the conversation ID that lets you "specify custom conversation" but my efforts in finding where to set this have proved fruitless.

A few questions; How would I generally and briefly set this so I can access a previous conversation? What should the ID be set to in a general use case.? Is it possible to call up previously established chats and if so, where would I find their ID be found?

Thank you very much for your time.

**Comment by Jamesjjvr** [comment]
After Further searching I have found what I assume to be the UUID, the string after the final slash in a url for a conversation. It looks something like this "conversation": "xxxxxxxx-8aac-41bd-b209-8e6f110bb282" Inputting this into the config makes the AI however output "

        ChatGPT - A command-line interface to OpenAI's ChatGPT (https://chat.openai.com/chat)
        Repo: github.com/acheong08/ChatGPT

Type '!help' to show commands
Press enter twice to submit your question.

Logging in...
Spawning browser...
Browser spawned.
Found Cloudflare Cookie!

You:

Chatbot:
{"detail":"Something went wrong, please try reloading the conversation."}
Something went wrong!
Wrong response code! Refreshing session...

You:


I am unsure if this is user error or something else. However as I note the AI worked fine before I added the UUID and works fine after I remove it, i am assuming user error. Your guidance and time remain appreciated.

**Reply by acheong08** [reply]
You also need to set parent_id along with conversation_id

**Reply by Jamesjjvr** [reply]
Thank you! Where would I go about doing this? I've put the conversation ID in the config as it has a preset option. Would I also put the parent ID in the config? Where would I find the parent ID?

**Reply by acheong08** [reply]
You can get parent_id programmatically by accessing the Chatbot.parent_id. Not sure how to get it from the website.
